src/urllib3/contrib/_appengine_environ.py

```diff
@@ -6,24 +6,21 @@
 
 
 def is_appengine():
-    return (is_local_appengine() or
-            is_prod_appengine() or
-            is_prod_appengine_mvms())
+    return 'APPENGINE_RUNTIME' in os.environ
 
 
 def is_appengine_sandbox():
     return is_appengine() and not is_prod_appengine_mvms()
 
 
 def is_local_appengine():
-    return ('APPENGINE_RUNTIME' in os.environ and
-            'Development/' in os.environ['SERVER_SOFTWARE'])
+    return is_appengine() and \
```

Comment:
This isn't going to work right. This will return `True` for Flex as `SERVER_SOFTWARE` won't be set but `APPENGINE_RUNTIME` will be. We either need to *require* `SERVER_SOFTWARE` as part of the local check or find a more reliable way to determine whether or not you're in the local environment.

Reply:
Why `is_local_appengine` needs to know weather it is an flex environment or not?
For this distinction we use `is_prod_appengine_mvms`.

Reply:
The alternative is to add `not is_prod_appengine_mvms` to `is_prod_appengine` and `is_local_appengine`

Reply:
Basically each of these checks needs to be definitive:

`is_local_appengine` should return True if and only if you're on the local app engine dev server.

`is_prod_appenegine` should return Tue if and only if you're in the production app engine (standard or flex).

and so on. Does that make sense?

Reply:
What about the import check?